app/containers/Initializer.js

```diff
@@ -48,74 +41,42 @@ class Initializer extends React.Component {
+   * or null if there is no location change
    */
-  componentDidUpdate(prevProps) {
+  getLocation() {
     const {
       activeWallet,
       activeWalletSettings,
       isReady,
       isWalletOpen,
       hasWallets,
-      history,
-      lightningGrpcActive,
-      lndConnect,
-      startLndError,
-      walletUnlockerGrpcActive,
-      startActiveWallet
+      lndConnect
     } = this.props
-    // If the wallet settings have just been loaded, redirect the user to the most relevant location.
-    if (isReady && !prevProps.isReady) {
-      // If we have an lndConnect link, send the user to the onboarding process.
-      if (lndConnect) {
-        this.nextLocation = '/onboarding'
-      }
-
-      // Otherwise, attempt to handle their current active wallet.
-      else if (activeWalletSettings) {
-        if (isWalletOpen) {
-          // Catch the error and swallow it with a noop.
-          // Errors are handled below by listening for updates to the startLndError prop.
-          return startActiveWallet().catch(() => {})
-        } else {
-          this.nextLocation = `/home/wallet/${activeWallet}`
-        }
-      }
 
-      // If we have at least one wallet send the user to the homepage.
-      // Otherwise send them to the onboarding processes.
-      else {
-        this.nextLocation = hasWallets ? '/home' : '/onboarding'
-      }
+    // still initializing - no location change
+    if (!isReady) {
+      return null
     }
 
-    // If there was a problem starting lnd, swich to the wallet launcher.
-    if (startLndError && !prevProps.startLndError) {
-      this.nextLocation = `/home/wallet/${activeWallet}`
+    // Came here from an lnd connect link
+    if (lndConnect) {
+      return '/onboarding'
     }
 
-    // If the wallet unlocker became active, switch to the login screen.
-    if (walletUnlockerGrpcActive && !prevProps.walletUnlockerGrpcActive) {
-      this.nextLocation = `/home/wallet/${activeWallet}/unlock`
+    // we have either an open wallet or active wallet settings
+    if (activeWalletSettings) {
+      return isWalletOpen ? '/wallet-starter' : `/home/wallet/${activeWallet}`
     }
 
-    // If an active wallet connection has been established, switch to the app.
-    if (lightningGrpcActive && !prevProps.lightningGrpcActive) {
-      if (activeWalletSettings.type === 'local') {
-        this.nextLocation = '/syncing'
-      } else {
-        this.nextLocation = '/app'
-      }
-    }
-
-    // Once the app is ready, and we have determined the next location, redirect the user.
-    if (isReady && this.nextLocation) {
-      return history.push(this.nextLocation)
-    }
+    // If we have at least one wallet send the user to the homepage.
+    // Otherwise send them to the onboarding processes.
+    return hasWallets ? '/home' : '/onboarding'
   }
 
   render() {
-    return null
+    const location = this.getLocation()
+    return location && <Redirect to={location} />
```

Comment:
I like the way you are doing this in the render function now. This simplifies things a lot and should also mean that it covers the case where the next location is already known when the component first mounts (previously, we only checked in the component update hook)